source/docs/software/commandbased/builtins.rst

```diff
@@ -0,0 +1,379 @@
+
+.. tabs::
+
+  .. code-tab:: java
+
+    new FunctionalCommand(
+        // Reset encoders on command start
+        m_robotDrive::resetEncoders,
+        // Start driving forward at the start of the command
+        () -> m_robotDrive.arcadeDrive(kAutoDriveSpeed, 0),
+        // Stop driving at the end of the command
+        interrupted -> m_robotDrive.arcadeDrive(0, 0),
+        // End the command when the robot's driven distance exceeds the desired value
+        () -> m_robotDrive.getAverageEncoderDistance() >= kAutoDriveDistanceInches,
+        // Require the drive subsystem
+        m_robotDrive
+    )
+
+  .. code-tab:: c++
+
+    frc2::FunctionalCommand(
+      // Reset encoders on command start
+      [this] { m_drive.ResetEncoders(); },
+      // Start driving forward at the start of the command
+      [this] { m_drive.ArcadeDrive(ac::kAutoDriveSpeed, 0); },
+      // Stop driving at the end of the command
+      [this] (bool interrupted) { m_drive.ArcadeDrive(0, 0); },
+      // End the command when the robot's driven distance exceeds the desired value
+      [this] { return m_drive.GetAverageEncoderDistance() >= kAutoDriveDistanceInches; },
+      // Requires the drive subsystem
+      {&m_drive}
+    )
+
+PrintCommand
+^^^^^^^^^^^^
+
+``PrintCommand`` (`Java <https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/PrintCommand.html>`__, `C++ <https://first.wpi.edu/wpilib/allwpilib/docs/release/cpp/classfrc2_1_1_print_command.html>`__) is a subclass of ``InstantCommand`` for printing a string and ending immediately.
+
+.. tabs::
+
+  .. code-tab:: java
+
+    new PrintCommand("This message will be printed!")
+
+  .. code-tab:: c++
+
+    frc2::PrintCommand("This message will be printed!")
+
+
+Dynamically Deciding What Command To Schedule
+---------------------------------------------
+
+Sometimes it's desired to run a command out of a few options based on sensor feedback or other data known only at runtime.
+
+SelectCommand
+^^^^^^^^^^^^^
+
+For this, ``SelectCommand`` (`Java <https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/SelectCommand.html>`__, `C++ <https://first.wpi.edu/wpilib/allwpilib/docs/release/cpp/classfrc2_1_1_select_command.html>`__) accepts a map of commands and a generic selector, or a ``Supplier<Command>``.
+
+.. note:: While the Java version of ``SelectCommand`` simply uses an ``Object`` as a key, the C++ version is templated on the key type.
+
+.. note:: An alternate version of ``SelectCommand`` simply takes a method that supplies the command to be run - this can be very succinct, but makes inferring the command’s requirements impossible, and so leaves the user responsible for manually adding the requirements to the SelectCommand.
```

Comment:
This sentence is long and hard to read. Can it be broken into multiple sentences.